test/smoke/src/sql/areas/notebook/notebook.test.ts

```diff
@@ -132,6 +133,51 @@ export function setup(opts: minimist.ParsedArgs) {
 			await app.code.dispatchKeybinding('escape');
 		});
 
+		describe('Notebook keyboard navigation', async () => {
+			it('can enter and exit edit mode and navigate using keyboard nav', async function () {
+				const app = this.app as Application;
+				await app.workbench.sqlNotebook.newUntitledNotebook();
+				await app.workbench.sqlNotebook.addCellFromPlaceholder('Code'); // add new code cell
+				await app.workbench.sqlNotebook.waitForPlaceholderGone();
+				const activeCodeCellId = (await app.workbench.sqlNotebook.getActiveCell()).attributes['id'];
+				await app.workbench.sqlNotebook.waitForTypeInEditor('code cell', activeCodeCellId); // the new cell should be in edit mode
+				await app.workbench.sqlNotebook.exitActiveCell();  // hitting escape twice deselects all cells
+				await app.code.dispatchKeybinding('escape');
+				await app.workbench.sqlNotebook.waitForActiveCellGone();
+
+				await app.workbench.sqlNotebook.addCell('markdown'); // add markdown cell
+				await app.workbench.sqlNotebook.textCellToolbar.changeTextCellView('Split View');
+				const activeTextCellId = (await app.workbench.sqlNotebook.getActiveCell()).attributes['id'];
+				await app.workbench.sqlNotebook.waitForTypeInEditor('text cell', activeTextCellId); // Text cell should be in edit mode
+
+				await app.code.dispatchKeybinding('escape'); // exit edit mode and stay in browse mode
+				await app.code.dispatchKeybinding('up'); // select code cell
+				await app.workbench.sqlNotebook.getActiveCell(activeCodeCellId); // check that the code cell is now active
+				await app.code.dispatchKeybinding('enter');
+				await app.workbench.sqlNotebook.waitForTypeInEditor('test', activeCodeCellId); // code cell should be in edit mode after hitting enter
+				await app.code.dispatchKeybinding('escape'); // exit edit mode and stay in browse mode
+				await app.code.dispatchKeybinding('down'); // select text cell
+				await app.code.dispatchKeybinding('enter');
+				await app.workbench.sqlNotebook.textCellToolbar.changeTextCellView('Split View');
+				await app.workbench.sqlNotebook.waitForTypeInEditor('test', activeTextCellId); // text cell should be in edit mode after hitting enter
+			});
+
+			it('cannot move through cells when find widget is invoked', async function () {
+				const app = this.app as Application;
+				await app.workbench.sqlNotebook.newUntitledNotebook();
```

Comment:
What I find helpful for stuff like this is to add comments for splitting up the sections into three parts : 

1. Setup
2. Act
3. Verify 

So in this case 166-173 are setup, 174-176 are act and 177 is verify. 

Something to keep in mind when you're writing tests, it makes them a lot easier for humans to read and understand.